corehq/apps/linked_domain/local_accessors.py

```diff
@@ -10,6 +10,40 @@
 from corehq.apps.users.models import SQLUserRole
 from corehq.apps.users.views.mobile import UserFieldsView
 from corehq.apps.integration.models import DialerSettings, GaenOtpServerSettings, HmacCalloutSettings
+from corehq.apps.reports.models import TableauServer, TableauVisualization
+
+
+def get_tableau_server_and_visualizations(domain):
+    server, created = TableauServer.objects.get_or_create(domain=domain)
+    visualizations = TableauVisualization.objects.all().filter(domain=domain).order_by('pk')
+    vis_list = []
+    if not len(visualizations):
```

Comment:
I don't think this is needed. If someone updates tableau models, and there are no visualizations in the upstream domain (unlikely, but possible), this'll create a visualization in the downstream domain, won't it?

I see that other linked domain code uses `get_or_create`, which has the same behavior, but those models are settings objects where there's only one for each domain, and making an empty/default object is okay.

Reply:
Ok so reach out for the visualizations and return those if they are there, but otherwise return an empty list?

Reply:
yes